torchao/quantization/subclass.py

```diff
@@ -191,19 +191,46 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
             )
 
 
+@torch._dynamo.allow_in_graph
+def from_qtensor_components_int8dyn(
+    *args,
+    **kwargs
+):
+    return Int8DynamicallyQuantizedLinearWeight(
+        *args,
+        **kwargs,
+    )
+
+class ConstructTensorSubclassInt8Dyn(torch.nn.Module):
+    def __init__(self, tensor_subclass_ctr, *args, **kwargs):
+        super().__init__()
+        self.tensor_subclass_ctr = tensor_subclass_ctr
+        self.args = args
+        self.kwargs = kwargs
+
+    def forward(self, int_data, q_scales):
+        return from_qtensor_components_int8dyn(int_data, q_scales, *self.args, **self.kwargs)
```

Comment:
Can you use tensor_unflatten for this?

Reply:
you mean tensor unflatten? we can't use `cls` in forward because of https://github.com/pytorch/pytorch/issues/124735 right now

Reply:
If you wrap do

```
def create_parameterization_module(cls):
    class SubclassParameterization:
        [...]

    def forward(self, args):
        cls.[...](args)

    return SubclassParameterization
```

then cls is given as an argument to `create_parameterization_module` and you return an instance of `SubclassParameterization` where `cls` is that argument. Essentially a module factory function.

These methods also shouldn't be static.

Reply:
isn't this using `cls` in forward? I tried this before, and with `@torch._dynamo.allow_in_graph` for the constructor function and it fails because we can't use class variable in dynamo right now I think.

are you suggesting something like this: https://github.com/pytorch/ao/commit/25abb31bfa57b67935ca30f77f9323d0a9345ec9#diff-bf4d50867e3d649de2d89146592bf47d2f258c4c19126c8acf0e120ee904b726R134 (but using `cls` instead of hardcoding the class?

Reply:
Yes exactly and using `__tensor_unflatten__` instead of `from_qtensor_components`

Reply:
for reference using `cls` in forward is not supported until https://github.com/pytorch/pytorch/pull/123350 is landed, according to Brain